### ISSUE TITLE
[Training] [M-IS] Phase B: Train Combined Force Student Model

### ISSUE THREAD

**Issue by atfrank** [atfrank/MLFF-Distiller#50]
## Overview
Train the final student model on combined forces (vacuum + solvation) to create a unified force field that includes implicit solvation effects.

## Background
**Phase B Goal**: A single student model that predicts total forces including solvation, enabling MD simulations with implicit solvent effects.

This model will:
- Accept same inputs as our current StudentForceField (atomic numbers, positions)
- Output total forces that include solvation effects
- Serve as a drop-in replacement for vacuum-only student models

### Option 1: Train from Scratch
Train new student model entirely on combined teacher data.
- **Pros**: Optimized for combined forces
- **Cons**: Longer training time, may lose vacuum accuracy

### Option 2: Fine-tune Existing Model
Start from trained vacuum student model and fine-tune on combined data.
- **Pros**: Faster convergence, retains vacuum accuracy
- **Cons**: May have trouble learning solvation component

### Option 3: Multi-Task Learning
Train single model to predict both vacuum and solvation components.
- **Pros**: Interpretable, can output components separately
- **Cons**: More complex architecture

### Loss Function
```python
# Option 1: Total force loss only
loss = MSE(predicted_total_forces, target_total_forces)

# Option 2: Multi-task
loss = (
    total_weight * MSE(predicted_total, target_total) +
    vacuum_weight * MSE(predicted_vacuum, target_vacuum) +
    solvent_weight * MSE(predicted_solvent, target_solvent)
)
```

### Hyperparameters (Starting Point)
- Model: StudentForceField (PaiNN-based)
- hidden_dim: 128
- num_interactions: 3
- Learning rate: 1e-4
- Batch size: 32
- Epochs: 100-200

### Setup
- [ ] Load combined teacher dataset from Phase B
- [ ] Configure training loop for combined forces
- [ ] Set up evaluation metrics
- [ ] Prepare comparison baselines

### Training
- [ ] Implement chosen training strategy
- [ ] Run training with validation monitoring
- [ ] Hyperparameter optimization
- [ ] Compare different training strategies

### Validation
- [ ] Evaluate total force accuracy (RMSE, MAE, R2)
- [ ] Compare vacuum force preservation
- [ ] Analyze solvation force accuracy
- [ ] Test on external validation molecules

### MD Validation
- [ ] Run test MD simulations with combined student
- [ ] Compare trajectories vs vacuum-only simulations
- [ ] Evaluate energy conservation
- [ ] Check for solvation-induced conformational changes

## Success Criteria
- Total Force R2 > 0.95
- Force RMSE < 0.2 eV/A
- MD energy drift < 1% over 10ps NVE
- Solvation effects qualitatively correct (polar molecules solvated)

## Deliverables
- [ ] Trained combined student model checkpoint
- [ ] Training logs and metrics
- [ ] Comparison with vacuum-only model
- [ ] MD validation results
- [ ] Documentation in `docs/training/COMBINED_STUDENT_TRAINING.md`

## Dependencies
- Depends on Issue #49 (Phase B Dataset)

## Agent Assignment
**Primary**: Training Agent
**Secondary**: Testing Agent (for MD validation)

## Priority
High
